### PR TITLE
REV: Set depth as default vertical_domain

### PR DESCRIPTION
**Pull request by tnatt**

### CODE REVIEW

File: src/fmu/dataio/dataio.py [hunk 1]
```diff
@@ -221,7 +221,7 @@ class ExportData:
             Note also: If missing or empty, export() may still be done, but without a
             metadata file (this feature may change in future releases).
 
-        content: Optional, default is "depth". Is a string or a dictionary with one key.
```
Comment (tnatt) [inline src/fmu/dataio/dataio.py:224]: Sneaked in a removal of incorrect documentation

**Comment by perolavsvendsen** [comment]
Associated discussion: https://equinor.slack.com/archives/C06E3UTGTEV/p1727263798090899

**Review by perolavsvendsen — APPROVED** [review]
LGTM, assuming tests will pass, thanks for fast response 🚀

**Comment by mferrera** [comment]
Force merging this as the only failing test is due to a change in sumo uploader that broke create case metadata

**Pull request merged**